tipg/dbmodel.py

```diff
@@ -507,24 +508,29 @@ def _where(
             wheres.append(to_filter(cql, [p.name for p in self.properties]))
 
         if tile and tms and geometry_column:
-            tilebox = tms.bounds(tile)
+            # Get Tile Bounds in Geographic CRS (usually epsg:4326)
+            left, bottom, right, top = tms.bounds(tile)
+
+            # Truncate bounds to the max TMS bbox
+            left, bottom = tms.truncate_lnglat(left, bottom)
+            right, top = tms.truncate_lnglat(right, top)
```

Comment:
to me it feels weird that we use bounds in geographic CRS instead of the bounds in TMS CRS. I guess the issue is that we cannot use `ST_MakeEnvelope` with non-epsg CRS 🤷

Reply:
The intersects should be in the datasets srs since that is what has a spatial index on it.

Reply:
sure, but at this line we use `bounds` in WGS84 from morecantile to construct the envelope which is then Reprojected to the geometry CRS. My comment is mostly to say that we shouldn't rely on morecantile bounds in WGS84, but do the transformation and creation of the envelop in Postgres